bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -182,13 +183,26 @@ protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHtt
                 component = componentCandidate;
             }
 
-
             ImageComponent imageComponent = new ImageComponent(component);
+            // If the image has no content, use the featured image of the page if it exists
             if (imageComponent.source == Source.NONEXISTING) {
-                LOGGER.error("The image from {} does not have a valid file reference.", component.getPath());
-                metrics.markImageErrors();
-                response.sendError(HttpServletResponse.SC_NOT_FOUND);
-                return;
+                PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
+                if (pageManager != null) {
+                    Page page = pageManager.getContainingPage(component);
+                    if (page != null) {
+                        Resource featuredImage = ComponentUtils.getFeaturedImage(page);
+                        if (featuredImage != null) {
+                            imageComponent = new ImageComponent(featuredImage);
+                            if (imageComponent.source == Source.NONEXISTING) {
```

Comment:
I would move this after the attempt of getting the `imageComponent` (Line 206) from the featured image, to cover the null-checks branching as well.

Reply:
Addressed in https://github.com/adobe/aem-core-wcm-components/pull/1626/commits/a5167ad95344a28b7008cb4e885be29ec99a5e2c